### PR TITLE
Fix Static Notches

### PR DESCRIPTION
**Pull request by andyp1per**
Bug introduced in https://github.com/ArduPilot/ardupilot/pull/25442
I will note in passing that my review indicated this shouldn't be removed.
I have added a test to make sure this can't regress

### CODE REVIEW

**Comment by IamPete1** [comment]
I think we should still remove the check here: https://github.com/ArduPilot/ardupilot/blob/3dd44dd8e07230f03c87302211498184dcdab6b3/libraries/AP_InertialSensor/AP_InertialSensor.cpp#L1871

It will allow runtime param changes to take affect for static notch as they do for the others.

**Comment by andyp1per** [comment]
> It will allow runtime param changes to take affect for static notch as they do for the others

This was/is covered by the init() above. The reason not to do it here is the cost of updating the static notch all the time when it doesn't need it.

**Comment by IamPete1** [comment]
> This was/is covered by the init() above.

The init allows runtime change in frequency/bandwidth/attenuation but not runtime change in harmonics or double/triple notch.

> The reason not to do it here is the cost of updating the static notch all the time when it doesn't need it.

I guess I don't really understand why we carefully check if we should re-init due to param change. It seems to be the update is the costly one we just do that all the time (except static). Why don't we check for a change in options or harmonic too and then call init and update at once?

**Comment by andyp1per** [comment]
> The init allows runtime change in frequency/bandwidth/attenuation but not runtime change in harmonics or double/triple notch.

I mean this is true, but I don't think what you are suggesting fixes this does it? I agree that allowing runtime changes to these properties would be convenient.


> I guess I don't really understand why we carefully check if we should re-init due to param change. It seems to be the update is the costly one we just do that all the time (except static). Why don't we check for a change in options or harmonic too and then call init and update at once?

Really just copying the pattern that was there for the LPF. It is true that calculating A&Q is expensive, so avoiding doing that all the time makes sense. However, we have made a lot of effort into making update cheap, so it may be that simply removing the check is not such a big deal anymore.

**Review by tridge — CHANGES_REQUESTED** [review]
good find, thanks!

File: libraries/Filter/HarmonicNotchFilter.cpp [hunk 1]
```diff
@@ -191,6 +191,8 @@ void HarmonicNotchFilter<T>::init(float sample_freq_hz, HarmonicNotchFilterParam
     NotchFilter<T>::calculate_A_and_Q(center_freq_hz, bandwidth_hz / _composite_notches, attenuation_dB, _A, _Q);
 
     _initialised = true;
+    // ensure static notches are allocated and working
+    update(center_freq_hz);
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/Filter/HarmonicNotchFilter.cpp:195]: fix here: https://github.com/tridge/ardupilot/commits/pr-static-notch/

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by andyp1per** [comment]
Thanks @tridge I've pulled in the change

**Review by tridge — APPROVED** [review]

**Review by IamPete1 — APPROVED** [review]
I think this is correct.

**Pull request merged**